enterprise/reporting/event_sampler/in_memory_cache_event_sampler.go

```diff
@@ -0,0 +1,56 @@
+package event_sampler
+
+import (
+	"sync"
+	"time"
+
+	"github.com/rudderlabs/rudder-go-kit/cachettl"
+	"github.com/rudderlabs/rudder-go-kit/config"
+)
+
+type InMemoryCacheEventSampler struct {
+	cache  *cachettl.Cache[string, bool]
+	mu     sync.Mutex
```

Comment:
we can get rid of the mu from what I can see.. cachettl already takes care of the safety